### PR TITLE
Enhanced Documentation

### PR DESCRIPTION
**Pull request by int0x81**
# New Pull Request

For more information, see the `CONTRIBUTING` guide.

## Prerequisites

Please make sure you can check the following boxes:

- [x] I have read the **CONTRIBUTING** document
- [x] My code follows the code style of this project
- [x] All new and existing tests passed

### Contribution Type

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue, please reference the issue id)
- [ ] New feature (non-breaking change which adds functionality, make sure to open an associated issue first)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes

### Description

This Pull request enhances the documentation according to issue #314 . The documentations for both official supported plugins are now beeing listed under an own section in the "Tooling" tab.

### CODE REVIEW

File: src/pages/docs/src/codegen/gen-bundlers.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+const { generated, getName, getBundlers } = require('./paths');
+const { render } = require('./markdown');
+const { generateStandardPage } = require('./pages');
+
+function getRoute(name) {
+  return (name && `/bundlers/${name}`) || '';
```
Comment (FlorianRappl) [inline src/pages/docs/src/codegen/gen-bundlers.js:6]: Can we have a `/tooling/...` path here - like for the other sections in tooling?

File: src/pages/docs/src/codegen/gen-bundlers.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+}
+
+module.exports = function() {
+  const bundlers = getBundlers();
+
+  const imports = bundlers.map(file => {
+
+    const { mdValue, meta = {} } = render(file, generated);
+    const nameToUse = meta.title.split(' ')[0];
```
Comment (FlorianRappl) [inline src/pages/docs/src/codegen/gen-bundlers.js:15]: I would not use `meta` here (more below).

Instead, the name could be inferred from the path, i.e., just take `piral-cli-webpack` / `piral-cli-parcel` (essentially just the plugin name).

File: src/tooling/piral-cli-parcel/README.md [hunk 3]
```diff
@@ -1,3 +1,7 @@
+---
```
Comment (FlorianRappl) [inline src/tooling/piral-cli-parcel/README.md:1]: Modifying this MD is not intended;

- it will be inconsistent to the other READMEs
- it will potentially misrender on GitHub / NPM / etc.

Same applies to the `piral-cli-webpack` README

**Review by FlorianRappl — COMMENTED** [review]
I think this is good so far.

File: src/pages/docs/src/codegen/gen-bundlers.js (hunk 1 above)
Comment (int0x81) [reply src/pages/docs/src/codegen/gen-bundlers.js]: adapted

File: src/pages/docs/src/codegen/gen-bundlers.js (hunk 2 above)
Comment (int0x81) [reply src/pages/docs/src/codegen/gen-bundlers.js]: Names are now beeing generated by the file path (e.g. `piral-cli-webpack` / `piral-cli-parcel`).

File: src/tooling/piral-cli-parcel/README.md (hunk 3 above)
Comment (int0x81) [reply src/tooling/piral-cli-parcel/README.md]: removed the meta tag 'title'

File: src/pages/docs/src/codegen/gen-bundlers.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+}
+
+module.exports = function() {
+  const bundlers = getBundlers();
+
+  const imports = bundlers.map(file => {
+
+    const { mdValue, meta = {} } = render(file, generated);
+    const pathElements = file.split('\\').length;
+    const nameToUse = file.split('\\')[pathElements - 2];
```
Comment (FlorianRappl) [inline src/pages/docs/src/codegen/gen-bundlers.js:16]: This is not a sound solution. The filename is OS dependent. Either you first normalize to `/` and then split on `/` or you use the OS Separator to perform the splitting.

Also you perform two split operations where only one should be. If you want to avoid a temp. variable just use `filter`:

```ts
const { sep } = require('path');
const [nameToUse] = file.split(sep).filter((_, i, arr) => i === arr.length - 2);
```

File: src/pages/docs/src/codegen/paths.js [hunk 5]
```diff
@@ -109,6 +109,18 @@ function getTools() {
   return getDocsFrom(tools);
 }
 
+function getBundlers() {
+
+  const toolingRoot = resolve(packagesRoot, 'tooling');
+  const webpackRoot = resolve(resolve(toolingRoot, 'piral-cli-webpack'));
+  const parcelRoot = resolve(resolve(toolingRoot, 'piral-cli-parcel'));
```
Comment (FlorianRappl) [inline src/pages/docs/src/codegen/paths.js:116]: Why `resolve(resolve(...))`? JUst a single `resolve` should it.

Comment (int0x81) [reply src/pages/docs/src/codegen/paths.js]: ups :stuck_out_tongue_winking_eye: , changed to use just one resolve

File: src/pages/docs/src/codegen/gen-bundlers.js (hunk 4 above)
Comment (int0x81) [reply src/pages/docs/src/codegen/gen-bundlers.js]: adapted

File: src/tooling/piral-cli-parcel/README.md [hunk 6]
```diff
@@ -1,6 +1,4 @@
-[![Piral Logo](https://github.com/smapiot/piral/raw/master/docs/assets/logo.png)](https://piral.io)
```
Comment (FlorianRappl) [inline src/tooling/piral-cli-parcel/README.md:1]: Why this change?

All our READMEs look similar. Compare:

https://raw.githubusercontent.com/smapiot/piral/develop/src/tooling/piral-cli/README.md
(same for `piral-cli-webpack`)

If the change was made due to the logo etc. being rendered (and looking ugly), please look at how plugins are rendered / done. You can customize this.

**Review by FlorianRappl — COMMENTED** [review]
I think we are nearly done here.

:beers:

**Comment by int0x81** [comment]
I also changed the README.md documentations for the 2 bundler plugins to be more aligned with the other tooling docs:

- removed the link to piral.io
- removed the Logo above the header

Ideas on that?

File: src/tooling/piral-cli-parcel/README.md (hunk 6 above)
Comment (int0x81) [reply src/tooling/piral-cli-parcel/README.md]: rolled back these 2 changes

**Review by FlorianRappl — APPROVED** [review]
LGTM - thanks :beers:!
